### PR TITLE
Failing the velero backup if pv is not supported

### PR DESCRIPTION
**Pull request by liuy1vmware**
Before this change, velero backup just skips the pv if the pv type is not supported
Afte this change, when the migration feature gate is enabled, an unsupported pv type marks the backup as PartiallyFailed

Signed-off-by: Yang Liu <liuy1@liuy1-a02.vmware.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Return error to make velero backup partially failed, if the pv is not a supported volume type
```

### CODE REVIEW

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/708/

**Comment by lintongj** [comment]
Just one nit on release note

```
Return error to make velero backup partially failed if the pv is not a supported volume type.
```

Change looks good. But don't forget to run through regression test.

**Review by lintongj — APPROVED** [review]

File: pkg/plugin/backup_pvc_action_plugin.go [hunk 1]
```diff
@@ -100,15 +100,25 @@ func (p *NewPVCBackupItemAction) Execute(item runtime.Unstructured, backup *vele
...
+	if isResticUsed {
+		p.Log.Infof("Skipping PVC %s/%s, PV %s will be backed up using restic", pvc.Namespace, pvc.Name, pv.Name)
+		return item, nil, nil
+	}
+
 	if utils.IsFeatureEnabled(kubeClient, constants.CSIMigratedVolumeSupportFlag, false, p.Log) {
 		if pv.Spec.PersistentVolumeSource.CSI == nil && !pluginUtil.IsMigratedCSIVolume(pv) {
 			p.Log.Infof("Skipping PVC %s/%s, associated PV %s is not a CSI volume, nor a migrated volume", pvc.Namespace, pvc.Name, pv.Name)
-			return item, nil, nil
+			return nil, nil, errors.New("Only vsphere CSI or vsphere CSI migrated volume is supported")
```
Comment (xing-yang) [inline pkg/plugin/backup_pvc_action_plugin.go:116]: I think we should print out PVC namespace, name and PV name in the error message as well. So I suggest format the message (combine message on line 115 and 116) first, then print it out in Infof, and use the same formatted message when returning the error.

Also the first letter of a returned error message should be lower case.

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/711/

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/712/

File: pkg/plugin/backup_pvc_action_plugin.go [hunk 2]
```diff
@@ -100,15 +100,27 @@ func (p *NewPVCBackupItemAction) Execute(item runtime.Unstructured, backup *vele
...
+			p.Log.Infof(msg)
+			return nil, nil, errors.New("unsupported pvc " + msg)
 		}
 
 		if pv.Spec.PersistentVolumeSource.CSI != nil && pv.Spec.PersistentVolumeSource.CSI.Driver != constants.VSphereCSIDriverName {
-			p.Log.Infof("Skipping PVC %s/%s, associated PV %s is not a vSphere CSI volume", pvc.Namespace, pvc.Name, pv.Name)
-			return item, nil, nil
+			msg := fmt.Sprintf("Skipping unsupported PVC %s/%s, associated PV %s is not a vSphere CSI volume", pvc.Namespace, pvc.Name, pv.Name)
+			p.Log.Infof(msg)
+			return nil, nil, errors.New("unsupported pvc " + msg)
```
Comment (xing-yang) [inline pkg/plugin/backup_pvc_action_plugin.go:123]: Same comment here.

**Comment by liuy1vmware** [comment]
> > After this change, when the migration feature gate is enabled, an unsupported pv type fails the whole velero backup
> 
> After this change, when the migration feature gate is enabled, an unsupported pv type marks the backup of that pv as failed
> 
> Have you tried this out? I believe Velero backup will be marked as "partial failure"?

Yes it will show as
NAME    STATUS            ERRORS   WARNINGS   CREATED                         EXPIRES   STORAGE LOCATION   SELECTOR
liuy2   PartiallyFailed   1        0          2022-04-05 17:48:24 +0000 UTC   23d       default            <none>

**Comment by xing-yang** [comment]
Can you update the PR description?

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/713/

**Comment by svcbot-qecnsdp** [comment]
Congratulations !! 

All the pre-check tests passed
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/716/

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/717/

**Comment by xing-yang** [comment]
LGTM

**Review by xing-yang — APPROVED** [review]

**Comment by svcbot-qecnsdp** [comment]
Congratulations !! 

All the pre-check tests passed
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/718/

**Pull request merged**